### DISCUSSION TITLE
Plugin has just stopped working without any reason

### DISCUSSION THREAD

**Discussion by AleksandrovichK** [community/community#47654] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

All of a sudden the icon of Copilot had changed to this:
![image](https://user-images.githubusercontent.com/28981661/219390621-53cc94af-6ffb-45c1-bfc8-b9bafc09867a.png)
and Copilot tab to this:
![image](https://user-images.githubusercontent.com/28981661/219391233-0ed5f5fa-ffd0-4e67-8c5e-0b13ec0e64aa.png)

And I've noticed that completions don't work anymore.

**Re-logging** doesn't help, it just says "Contact support". So here I am
**Re-installation** of the plugin doesn't help
**Changing the channel** from `nightly` to `stable` doesn't help

<hr>

Country: Poland
OS: MacOS 13.2
IDE: PyCharm 2022.3
Plugin version: Copilot 1.2.3.2385

**Comment by tuves** [comment, marked as answer]
Thank you for reporting this. The bug fix has now been deployed to production and affected users should be able to connect. 

Due to the recency of this incident, we are still investigating the contributing factors and will provide a more detailed update on cause and remediation in the February Availability Report, which we will publish the first Wednesday of March.

We sincerely apologize for any stress and/or frustration this has caused.

**Comment by akaihola** [comment]
I too can't make the plugin connect. I installed the newest plugin on
`PyCharm 2022.1 (Community Edition) Build #PC-221.5080.212, built on April 12, 2022`.
This is on NixOS Linux. When I try to connect using `Tools > GitHub Copilot > Login to GitHub` or the toolbar icon, I immediately get
![kuva](https://user-images.githubusercontent.com/13725/220345926-aa371d36-a0fd-4731-956b-c9d5fcd9e402.png)
and
![kuva](https://user-images.githubusercontent.com/13725/220345991-724a258c-0b2f-415e-aa96-9d5beb422d02.png)

I also tried three previous versions of the plugin, and they show the same behavior. I also successfully logged in to GitHub in `Settings > Version Control > GitHub`, but that doesn't seem to affect Copilot.

Copilot does work just fine on the same laptop with VSCode and the Copilot extensions.

There seem to be several reports of this on the JetBrains Marketplace [GitHub Copilot page](https://plugins.jetbrains.com/plugin/17718-github-copilot/reviews) as well.

**Reply by akaihola** [reply]
(see below for how I solved this)

**Comment by tuves** [comment]
We became aware that Copilot was unavailable in several countries following our efforts to restore access to Ukraine, and on further investigation, we identified the cause as originating from a change we deployed to block Copilot in trade-restricted countries. To prevent this from happening again, we've updated our reporting dashboards to better account for error rates due to trade restrictions and are implementing more robust testing of the trade restriction logic.

**Reply by akaihola** [reply]
I'm in Finland, but could trade restrictions still have a side effect on me? I suppose no, since the VSCode extension still works fine.

**Reply by akaihola** [reply]
FYI, I also tried activating a VPN with my traffic appearing from USA, Germany or Sweden, always restarted PyCharm and still saw Copilot failing to connect. Would this prove that the problem is not in trade restrictions?

**Reply by akaihola** [reply]
I now found the reason for the plugin not working for me, and was able to fix it. I'm using NixOS, and the environment in which PyCharm is run is non-standard in a way which prevents `copilot-agent-linux` from running. I created an FHS ([Filesystem Hierarchy Standard](https://wiki.linuxfoundation.org/lsb/fhs)) compliant environment instead, and the plugin started to work.

For benefit of other NixOS users, here's my `pycharm-fhs.nix` which I run with `nix-shell -p pycharm-fhs.nix`:
```nix
{ pkgs ? import <nixos-unstable> {} }:

(pkgs.buildFHSUserEnv {
  name = "pycharm";
  targetPkgs = pkgs: (with pkgs; [
    black
    isort
    jetbrains.pycharm-community
    python311
    zlib  # needed for NumPy
  ]);
  runScript = "pycharm-community";
}).env
```

**Reply by liff** [reply]
On NixOS you can alternatively [enable nix-ld](https://search.nixos.org/options?channel=23.05&show=programs.nix-ld.enable).
